packages/react-jss/src/createUseStyles.js

```diff
@@ -25,17 +25,20 @@ type CreateUseStyles = <Theme: {}>(Styles<Theme>, HookOptions<Theme> | void) =>
 const createUseStyles: CreateUseStyles = <Theme: {}>(styles, options = {}) => {
   const {index = getSheetIndex(), theming, name, ...sheetOptions} = options
   const ThemeContext = (theming && theming.context) || DefaultThemeContext
+
+  /* eslint-disable no-unused-vars */
   const useTheme =
     typeof styles === 'function'
       ? // $FlowFixMe[incompatible-return]
-        (): Theme => React.useContext(ThemeContext) || noTheme
+        (propsTheme?: Theme): Theme => propsTheme || React.useContext(ThemeContext) || noTheme
       : // $FlowFixMe[incompatible-return]
-        (): Theme => noTheme
+        (_?: Theme): Theme => noTheme
```

Comment:
I'm assuming you've added the unused `_` argument because `useTheme` gets passed `data.theme` on line 41 and you wanted to avoid any type errors? Would it be better to make the whole declaration a one liner in this case?

Something like
```ts
const useTheme = (theme?: Theme): Theme => typeof styles !== 'function' ? noTheme : theme || React.useContext(ThemeContext) || noTheme;
```

They're both the same type of function, so there should only be a need to switch between the return types (instead of switching between function declarations).

Reply:
addressed in 1b78ac483c267642ad445bdbefc49790feae0c83